dagshub/data_engine/model/datasource.py

```diff
@@ -91,13 +91,13 @@ def to_dict(self, ds: "Datasource") -> Dict[str, Any]:
 
 
 class Datasource:
-    def __init__(self, datasource: "DatasourceState", query: Optional[DatasourceQuery] = None):
+    def __init__(self, datasource: "DatasourceState", query: Optional[DatasourceQuery] = None, select=[], as_of=None):
```

Comment:
Python annoying thing: you can't have mutable default values, because it's going to be a reference.
That means that if you append to the select, it might change values that you create later.
Better to have the select=None and in the constructor init it to [] if it's None

Reply:
oh yes, i should have known i ate shit from this in the past